linkml/generators/excelgen.py

```diff
@@ -129,48 +117,49 @@ def column_enum_validation(
 
         dv.add(f"{column_letter}2:{column_letter}1048576")
 
-        workbook_name = self.get_workbook_name(workbook)
-        workbook.save(workbook_name)
-
     def serialize(self, **kwargs) -> str:
-        self.output = (
-            os.path.abspath(convert_to_snake_case(self.schema.name) + ".xlsx") if not self.output else self.output
-        )
+        sv = self.schemaview
+        classes_to_process = [
+            cls_name
+            for cls_name, cls in sv.all_classes(imports=self.mergeimports).items()
+            if not cls.mixin and not cls.abstract
+        ]
 
-        workbook = self.create_workbook(self.output)
-        self.remove_worksheet_by_name(workbook, "Sheet")
-        self.create_schema_worksheets(workbook)
+        if self.split_workbook_by_class:
+            self.output = os.path.abspath(self.schema.name + "_worksheets") if not self.output else self.output
 
-        sv = self.schemaview
-        for cls_name, cls in sv.all_classes(imports=self.mergeimports).items():
-            if not cls.mixin and not cls.abstract:
-                slots = [s.name for s in sv.class_induced_slots(cls_name, imports=self.mergeimports)]
-                self.add_columns_to_worksheet(workbook, cls_name, slots)
+            os.makedirs(self.output, exist_ok=True)
```

Comment:
Do you use still os or has linkML switched to pathlib?

Reply:
I think the code is still a mix but I prefer using pathlib for cases like this.

Also: good idea to do parents=True too